Activation mode option

adrelino opened the pull request:
Adds the activationMode 'click'/'mouseover' option.

Its necessity was discussed in many issues already.

The defaults are set so that nothing changes the current behavior, just have a look at the `examples/layerswitcher.js` where I added an option to the top to switch between the two different modes.

walkermatt commented:
Demo: https://raw.githack.com/adrelino/ol-layerswitcher/activation-mode-option/examples/layerswitcher.html

walkermatt commented:
Thanks for the PR, happy to merge with a couple of changes:

1. The border on the main button isn't displaying correctly when the activation mode is click
2. I'd be in favour of hiding the button's background image when the panel is open, just showing the `collapseLabel`

adrelino commented:
> Thanks for the PR, happy to merge with a couple of changes:
> 
> 1. The border on the main button isn't displaying correctly when the activation mode is click

you'r right, I will check this

> 2. I'd be in favour of hiding the button's background image when the panel is open, just showing the `collapseLabel`

I actually did this via .css in an additional stylesheet of mine via:
```css
.layer-switcher.shown button{
  background-image: unset;
}
```
since I did not want to modify your original css. But I can also add the `:not(.activationModeClick)` to your original .layer-switcher button css.

> 3. Can you please remove the select from the default example and instead create a separate example for activation mode

sure

adrelino commented:
Ok, I incorporated the requested changes from your points 1., 2. and 3., rebased on top of your current master and squashed all my commits into one. Should be ready for merge now.

adrelino commented:
Demo: https://raw.githack.com/adrelino/ol-layerswitcher/activation-mode-option/examples/activation-mode-click.html

walkermatt commented:
Looks great, thanks for your work on this :small_airplane:
